### ISSUE TITLE
[DocDB][LST] Packed Columns: ERROR:  Query error: Write response count mismatch

### ISSUE THREAD

**Issue by def-** [yugabyte/yugabyte-db#12880]
Jira Link: [DB-2632](https://yugabyte.atlassian.net/browse/DB-2632)
### Description

On a universe created with `  bin/yb-ctl --replication_factor 3 create --tserver_flags=ysql_enable_packed_row=true,ysql_packed_row_size_limit=1700 --master_flags=ysql_enable_packed_row=true,ysql_packed_row_size_limit=1700
`
Might be related to #12879
@spolitov Using [LST](https://github.com/yugabyte/yb-long-system-test/) against current master state (50cbd58f6bd317120fc2b7ebb4f0d83a4bceea27) fails with internal errors:
```
2022-06-14 08:41:07,736 MainThread INFO
2022-06-14 08:41:07,737 MainThread INFO     --------------------------------------------------------------------------------
2022-06-14 08:41:07,737 MainThread INFO     Running Long System Test 0.1
2022-06-14 08:41:07,737 MainThread INFO     --------------------------------------------------------------------------------
2022-06-14 08:41:07,737 MainThread INFO
2022-06-14 08:41:07,745 MainThread INFO     Reproduce with: git checkout e42027ff && ./long_system_test.py --nodes=127.0.0.1:5433,127.0.0.2:5433,127.0.0.3:5433 --threads=8 --runtime=0 --complexity=full --seed=159817
2022-06-14 08:41:08,007 MainThread INFO     Database version: PostgreSQL 11.2-YB-2.15.1.0-b0 on x86_64-pc-linux-gnu, compiled by clang version 12.0.1 (https://github.com/yugabyte/llvm-project.git bdb147e675d8c87cee72cc1f87c4b82855977d94), 64-bit
2022-06-14 08:41:08,009 MainThread INFO     Creating tables for database db_lst_159817
2022-06-14 08:41:24,505 MainThread INFO     Starting worker_0: CreateIndexAction, DropIndexAction, SetConfigAction, AddColumnAction
2022-06-14 08:41:24,506 MainThread INFO     Starting worker_1: SingleInsertAction, SingleUpdateAction, SingleDeleteAction, BulkInsertAction, BulkUpdateAction, SetConfigAction
2022-06-14 08:41:24,507 MainThread INFO     Starting worker_2: RandomSelectAction, SetConfigAction
2022-06-14 08:41:24,509 MainThread INFO     Starting worker_3: RandomSelectAction, SetConfigAction
2022-06-14 08:41:24,510 MainThread INFO     Starting worker_4: RandomSelectAction, SetConfigAction
2022-06-14 08:41:24,511 MainThread INFO     Starting worker_5: CreateIndexAction, DropIndexAction, SetConfigAction, AddColumnAction
2022-06-14 08:41:24,512 MainThread INFO     Starting worker_6: RandomSelectAction, SetConfigAction
2022-06-14 08:41:24,513 MainThread INFO     Starting worker_7: RandomSelectAction, SetConfigAction
2022-06-14 08:41:34,106 worker_1   ERROR    Unexpected query failure: InternalError_
Query: INSERT INTO tg1_0 (c0_text, c1_jsonb) VALUES ('58', '{"a": 9, "b": ["0"], "c": false}'::jsonb);
  values: None
  runtime: 2022-06-14 08:41:34.101 - 2022-06-14 08:41:34.106
  supports explain: True
  supports rollback: True
  affected rows: None
Action: SingleInsertAction
Error class: InternalError_
Error code: XX000
Error message: ERROR:  Query error: Write response count mismatch
Transaction isolation level: read uncommitted
DB Node: host: 127.0.0.2, port: 5433
DB Backend PID: 186056
```
Logging looks something like this, happened many times:
```
E0614 08:56:56.080338 192119 batcher.cc:609] Batcher (0x000000001388ec18), session (0x00000000113afb38): Received wrong number of responses compared to request(s) sent.
E0614 08:58:08.070195 179400 write_query.cc:666] Wrong number or mismatches: 1 vs 2
E0614 08:58:08.070262 179400 async_rpc.cc:618] Write response count mismatch: 0 Redis requests sent, 0 responses received. 0 Apache CQL requests sent, 0 responses received. 2 PostgreSQL requests sent, 1 responses received.
E0614 08:58:08.070281 179400 async_rpc.cc:626] Illegal state (yb/client/async_rpc.cc:625): Write response count mismatch, request: tablet_id: "ddbc9811e06248acb6563538e617df47" propagated_hybrid_time: 6779687272735277056 include_trace: false write_batch { transaction { transaction_id: "\305\260\331-$bBL\206\202\024D\266x\206J" isolation: SNAPSHOT_ISOLATION status_tablet: "520abb9ec87c4b59bee73c57493f5c14" priority: 11136240255342564422 start_hybrid_time: 6779687272735170560 locality: GLOBAL } DEPRECATED_may_have_metadata: true } read_time { read_ht: 6779687272735264768 DEPRECATED_max_of_read_time_and_local_limit_ht: 6779687274783264768 global_limit_ht: 6779687274783264768 in_txn_limit_ht: 18446744073709551615 local_limit_ht: 6779687274783264768 } pgsql_write_batch { client: YQL_CLIENT_PGSQL stmt_id: 94232672 stmt_type: PGSQL_INSERT table_id: "0000450a00003000800000000000453d" schema_version: 14 ybctid_column_value { value { binary_value: "S\216\363\177\024K\345E\242\205\010\323HX\226\340\327\000\000!" } } column_values { column_id: 1 expr { value { double_value: -63.219470959047122 } } } column_values { column_id: 2 expr { value { float_value: 31.7553101 } } } column_values { column_id: 3 expr { value { double_value: 27.851813552206124 } } } column_values { column_id: 4 expr { value { binary_value: "H\017\000\000I\256\377\377W\n\000\000\002" } } } column_values { column_id: 5 expr { value { binary_value: "H\017\000\0006\367\377\377\330\t\000\000\002" } } } column_values { column_id: 6 expr { value { binary_value: "H\017\000\000\016\227\377\377\352\337\377\377\002" } } } column_values { column_id: 7 expr { value { binary_value: "B\017\000\000\033\200\2079\000\005\036U\031\001\005`\023\033\000\207D\000d\004\202\026\234\024 \003\002" } } } column_values { column_id: 8 expr { value { float_value: 46.837944 } } } column_refs { } ysql_catalog_version: 994 partition_key: "" } pgsql_write_batch { client: YQL_CLIENT_PGSQL stmt_id: 51843168 stmt_type: PGSQL_UPSERT table_id: "0000450a0000300080000000000045b5" schema_version: 0 range_column_values { value { float_value: 46.837944 } } range_column_values { value { float_value: 31.7553101 } } range_column_values { value { double_value: 27.851813552206124 } } range_column_values { value { double_value: -63.219470959047122 } } range_column_values { value { binary_value: "S\216\363\177\024K\345E\242\205\010\323HX\226\340\327\000\000!" } } column_refs { } ysql_catalog_version: 994 partition_key: "" } client_id1: 2399774823823658397 client_id2: 17313700842916508570 request_id: 1170 min_running_request_id: 1170 rejection_score: 0 batch_idx: 0, response: propagated_hybrid_time: 6779687272735772672 pgsql_response_batch { }
```
[lst.zip](https://github.com/yugabyte/yugabyte-db/files/8898021/lst.zip)

**Comment by bmatican** [comment]
@def- Can you re-run this and confirm if Sergei's diff above fixes this?

**Comment by def-** [comment]
Will do so now.

**Comment by bmatican** [comment]
@def- any repro, after Sergei's fix? if not, can we close this? thanks!

**Comment by def-** [comment]
Not seen in 6 days, good enough to close.

**Comment by def-** [comment]
@spolitov Should this be fixed in 2.14.2.0? I just got the same error on YB Managed and noticed your change is not included in 2.14 branch:
```
[dfelsing@dev-server-dfelsing yb-long-system-test]$ python3.9 ./long_system_test.py --nodes=X:5433 --threads=10 --runtime=60 --max-columns=20 --complexity=full --username admin --password "X"
2022-09-28 21:30:45,624 MainThread INFO
2022-09-28 21:30:45,624 MainThread INFO     --------------------------------------------------------------------------------
2022-09-28 21:30:45,624 MainThread INFO     Running Long System Test 0.1 - https://github.com/yugabyte/yb-long-system-test/
2022-09-28 21:30:45,625 MainThread INFO     --------------------------------------------------------------------------------
2022-09-28 21:30:45,625 MainThread INFO
2022-09-28 21:30:53,603 MainThread INFO     Reproduce with: git checkout f76f4c65 && ./long_system_test.py --nodes=X:5433 --threads=10 --runtime=60 --max-columns=20 --complexity=full --username admin --password X --seed=625394
2022-09-28 21:30:54,091 MainThread INFO     Database version: PostgreSQL 11.2-YB-2.14.2.0-b0 on x86_64-pc-linux-gnu, compiled by clang version 12.0.1 (https://github.com/yugabyte/llvm-project.git bdb147e675d8c87cee72cc1f87c4b82855977d94), 64-bit
2022-09-28 21:30:54,095 MainThread INFO     Creating tables for database db_lst_625394
2022-09-28 21:31:24,184 MainThread INFO     Starting worker_0: SingleInsertAction, SingleUpdateAction, SingleDeleteAction, BulkInsertAction, BulkUpdateAction, SetConfigAction
2022-09-28 21:31:24,185 MainThread INFO     Starting worker_1: RandomSelectAction, SetConfigAction
2022-09-28 21:31:24,186 MainThread INFO     Starting worker_2: SingleInsertAction, SingleUpdateAction, SingleDeleteAction, BulkInsertAction, BulkUpdateAction, SetConfigAction
2022-09-28 21:31:24,187 MainThread INFO     Starting worker_3: RandomSelectAction, SetConfigAction
2022-09-28 21:31:24,188 MainThread INFO     Starting worker_4: RandomSelectAction, SetConfigAction
2022-09-28 21:31:24,189 MainThread INFO     Starting worker_5: SingleInsertAction, SingleUpdateAction, SingleDeleteAction, BulkInsertAction, BulkUpdateAction, SetConfigAction
2022-09-28 21:31:24,190 MainThread INFO     Starting worker_6: RandomSelectAction, SetConfigAction
2022-09-28 21:31:24,190 MainThread INFO     Starting worker_7: CreateIndexAction, DropIndexAction, SetConfigAction, AddColumnAction
2022-09-28 21:31:24,191 MainThread INFO     Starting worker_8: RandomSelectAction, SetConfigAction
2022-09-28 21:31:24,192 MainThread INFO     Starting worker_9: CreateIndexAction, DropIndexAction, SetConfigAction, AddColumnAction
2022-09-28 21:31:34,202 MainThread INFO     Worker queries/s: [001.8][001.8][002.1][000.7][001.9][006.4][001.1][000.3][000.5][000.6]
2022-09-28 21:31:44,210 MainThread INFO     Worker queries/s: [002.3][002.3][002.1][000.7][003.4][007.2][002.7][000.5][000.8][000.4]
2022-09-28 21:31:54,220 MainThread INFO     Worker queries/s: [002.0][003.3][002.6][000.4][003.1][002.1][003.4][000.2][001.0][000.2]
2022-09-28 21:32:04,225 MainThread INFO     Worker queries/s: [003.9][002.6][001.9][000.7][002.3][007.6][002.7][000.3][000.6][000.3]
2022-09-28 21:32:14,235 MainThread INFO     Worker queries/s: [003.3][003.3][003.1][000.8][003.4][008.3][003.6][000.4][001.2][000.4]
2022-09-28 21:32:24,246 MainThread INFO     Worker queries/s: [005.5][005.6][005.0][001.8][005.0][010.8][004.3][000.2][001.4][000.3]
2022-09-28 21:32:34,255 MainThread INFO     Worker queries/s: [003.4][004.6][003.9][000.7][003.3][011.3][004.2][000.2][000.9][000.1]
2022-09-28 21:32:44,264 MainThread INFO     Worker queries/s: [002.9][003.0][003.4][001.5][003.6][009.6][003.4][000.6][000.9][000.6]
2022-09-28 21:32:54,274 MainThread INFO     Worker queries/s: [004.6][004.6][005.3][000.7][004.7][009.4][002.8][000.2][000.8][000.4]
2022-09-28 21:33:04,284 MainThread INFO     Worker queries/s: [003.4][003.7][003.4][000.9][003.2][006.1][003.4][000.8][001.1][000.7]
2022-09-28 21:33:14,295 MainThread INFO     Worker queries/s: [002.9][002.5][003.4][000.7][002.6][007.8][002.5][000.6][000.6][000.5]
2022-09-28 21:33:24,305 MainThread INFO     Worker queries/s: [003.5][004.1][002.5][001.4][003.7][012.8][004.0][000.3][000.6][000.3]
2022-09-28 21:33:33,554 worker_5   ERROR    Unexpected query failure: InternalError_
Query: EXPLAIN (ANALYZE TRUE, VERBOSE FALSE, COSTS FALSE, BUFFERS TRUE, TIMING TRUE, SUMMARY FALSE, FORMAT XML)
DELETE FROM tg1_0 WHERE TRUE;
  values: None
  runtime: 2022-09-28 21:33:33.506 - 2022-09-28 21:33:33.553
  supports explain: False
  supports rollback: False
  affected rows: None
Action: SingleDeleteAction
Error class: InternalError_
Error code: XX000
Error message: ERROR:  Query error: Write response count mismatch
Transaction isolation level: read uncommitted
DB Node: host: 54.191.223.51, port: 5433
DB Backend PID: 28463
```

**Comment by bmatican** [comment]
@def- that's strange, packed was not available in 2.14, so this shouldn't be related...

**Comment by def-** [comment]
Could we also get schema version mismatches without packed columns in this case?
